server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -2211,6 +2214,31 @@ else if (details != null && !details.isEmpty()) {
         return _tmpltDao.findById(id);
     }
 
+    void validateDetails(VMTemplateVO template, Map<String, String> details) {
+        if (details == null || details.isEmpty()) {
+            return;
+        }
+        String bootMode = details.get(ApiConstants.BootType.UEFI.toString());
+        if (bootMode == null) {
+            return;
+        }
+        if (template.isDeployAsIs()) {
+            String msg = "Deploy-as-is templates can not have UEFI parameter set. Settings are read directly from the template";
+            throw new InvalidParameterValueException(msg);
```

Comment:
I think we shouldn't fail at this point, but ignore the settings (return at this point in the method)

Reply:
This might lead the user to wonder why the setting was not applied

Reply:
Yes, sorry, was thinking about the deployment but this is updating the settings +1